### ISSUE TITLE
Inviting comments on Connection Profiles concept

### ISSUE THREAD

**Issue by joshooaj** [MilestoneSystemsInc/PowerShellSamples#101]
The `Connect-ManagementServer` command is a bit long in the tooth and I'm thinking about introducing the concept of "connection profiles".

In some other PowerShell modules and CLI tools, you might connect to several different tenants depending on the type of work you're doing. So those modules often provide a way to easily switch between "profiles" or "contexts" without having to re-enter the credentials each time.

Many users of the MilestonePSTools module also find themselves connecting to different Milestone sites, or they may want to setup a scheduled job in Windows to periodically run a report against one or more Milestone servers without having to manage the storage and retrieval of credentials.

In the video below I demonstrate a few new commands including `Connect-Vms` which is intended to be a replacement for `Connect-ManagementServer`, and `Get-VmsConnectionProfile`, `Set-VmsConnectionProfile`, and `Remove-VmsConnectionProfile` which allow you to inspect, update, and remove named connection profiles.

```powershell
# Attempts to connect to the connection profile named "default".
# If the profile doesn't exist, a login dialog will be presented, and after successful logon the credentials are encrypted and saved to `~\AppData\Local\MilestonePSTools\credentials.xml` using `Export-CliXml`.
# Also new with this command is that it returns the result of `Get-VmsManagementServer` so that when used interactively it's clear that you successfully logged in, and you can confirm which server you're connected to.
Connect-Vms

# Connects to the management server on localhost, and any child sites if available, using the current Windows user credential.
# Then the current connection is saved as "local", and as a test, we reconnect using the connection profile named "local".
Connect-Vms -ServerAddress http://localhost -IncludeChildSites
Set-VmsConnectionProfile -Name local
Connect-Vms -Name local

# Returns a list of all saved connection profiles
Get-Vms -All
Get-VmsConnectionProfile -All

# If currently logged in to a management server and you want to make sure this is the server used in your default connection profile, you can use the -Force switch to overwrite the specified profile if it already exists.
Set-VmsConnectionProfile -Force

# Remove a specific connection profile, or remove them all.
Remove-VmsConnectionProfile local
Remove-VmsConnectionProfile -All
```

https://github.com/MilestoneSystemsInc/PowerShellSamples/assets/7669972/d1043fa9-5e1d-4227-8dcb-0e378ced1f4f

**Comment by Tarterman** [comment]
I like what you've done.  I see a lot of benefits.

**Comment by joshooaj** [comment]
Thanks @Tarterman, this is released with v23.2.2 so I'll close this out!
